common/v2/components/AccountList.tsx

```diff
@@ -46,45 +82,98 @@ export default function AccountList(props: AccountListProps) {
       actionLink="/add-account"
       className={`AccountList ${className}`}
     >
-      <CollapsibleTable
-        breakpoint={450}
-        {...buildAccountTable(currentsOnly ? currentAccounts : accounts, deleteAccount)}
-      />
+      <TableContainer>
+        <CollapsibleTable
+          breakpoint={450}
+          {...buildAccountTable(
+            currentsOnly ? currentAccounts : accounts,
+            deleteAccount,
+            updateAccount,
+            deletable
+          )}
+        />
+      </TableContainer>
     </DashboardPanel>
   );
 }
 
-function buildAccountTable(accounts: ExtendedAccount[], deleteAccount: DeleteAccount) {
+function buildAccountTable(
+  accounts: ExtendedAccount[],
+  deleteAccount: DeleteAccount,
+  updateAccount: UpdateAccount,
+  deletable?: boolean
+) {
   return {
-    head: [
-      translateRaw('ACCOUNT_LIST_FAVOURITE'),
-      translateRaw('ACCOUNT_LIST_ADDRESS'),
-      translateRaw('ACCOUNT_LIST_NETWORK'),
-      translateRaw('ACCOUNT_LIST_VALUE'),
-      translateRaw('ACCOUNT_LIST_DELETE')
-    ],
+    head: deletable
+      ? [
+          translateRaw('ACCOUNT_LIST_FAVOURITE'),
+          translateRaw('ACCOUNT_LIST_LABEL'),
+          translateRaw('ACCOUNT_LIST_ADDRESS'),
+          translateRaw('ACCOUNT_LIST_NETWORK'),
+          translateRaw('ACCOUNT_LIST_VALUE'),
+          translateRaw('ACCOUNT_LIST_DELETE')
+        ]
+      : [
+          translateRaw('ACCOUNT_LIST_FAVOURITE'),
+          translateRaw('ACCOUNT_LIST_LABEL'),
+          translateRaw('ACCOUNT_LIST_ADDRESS'),
+          translateRaw('ACCOUNT_LIST_NETWORK'),
+          translateRaw('ACCOUNT_LIST_VALUE')
+        ],
     body: accounts.map((account, index) => {
       const addressCard: AddressBook | undefined = getLabelByAccount(account);
       const label = addressCard ? addressCard.label : 'Unknown Account';
       let bodyItemCount = 0;
-      return [
-        <Icon key={index + bodyItemCount++} icon="star" />,
-        <Address
-          key={index + bodyItemCount++}
-          title={`${label}`}
-          address={account.address}
-          truncate={truncate}
-        />,
-        <Network key={index + bodyItemCount++} color="#a682ff">
-          {account.network}
-        </Network>,
-        <Typography key={index + bodyItemCount++}>{account.balance}</Typography>,
-        <DeleteButton
-          key={index + bodyItemCount++}
-          onClick={handleAccountDelete(deleteAccount, account.uuid)}
-          icon="exit"
-        />
-      ];
+      return deletable
+        ? [
+            <FavoriteButton
+              key={index + bodyItemCount++}
```

Comment:
declare `key` once

Reply:
is the key being declared more than once?

Reply:
Sorry for the delay to reply: 
`key={index + bodyItemCount++}` is set on every component in the `body`.
Though it would be better to have a 'row' component, let's avoid inline in incrementation and define a single key per row, something like `const rowIndex = index +...    key={rowIndex}`
I haven't tested it but I suspect `bodyItemCount++` to be entirely superfluous and that the array index is enough, so we can just do `key={index}`.

Reply:
Ahh that makes sense. I looked into it and `bodyItemCount` wasn't needed so I removed it and used index as suggested.